### ISSUE TITLE
Button a11y: Issues with the use of the component in storybook

### ISSUE THREAD

**Issue by khmakoto** [microsoft/fluentui#21542]
We found the following problems related to the use of the `Button` in storybook during the `Button` a11y review (#20917):
- We should remove the word "button" from the text so that it is not read twice by screen readers.
- We should add labels to all icon-only buttons.
- We should document that all the text in the `CompoundButton` is part of its name and hence should be concise.
